### ISSUE TITLE
Modal - keyboard navigation does not work when using the <a> anchor tag as the trigger button

### ISSUE THREAD

**Issue by emage** [twbs/bootstrap#28718]
When i switch the modal trigger from the <button> to a <a>, the modal does not respond to keyboard navigation. Tab focus is only applicable if a tabindex="0" is applied to it.

Is this just a usage pattern? I would prefer to still use the data-target for targeting the modal while using the <a> anchor tag as the trigger and not have to include a blank href="#". The dev team uses that href="#" on some other script.

Demo: https://codepen.io/Emage/pen/vMoyGV

Scenario:

Anchor tag using the default data-target
- tab is inaccessible - requires either a blank href or a tabindex="0" attribute to work
- keyboard trigger launch does not work
-  mouse trigger works

Anchor tag using href instead of data-target
- keyboard tab works
- keyboard launch works

**Comment by patrickhlauke** [comment]
if you are using `<a>` without an `href="..."`, the anchor won't be focusable by default, and yes you'll have to either add an empty `href="#"` or `tabindex="0"` (and make sure to add `role="button"` and ideally extra keyboard handling to make it react to `Space` activation). in general, it's not a recommended pattern...the best and most correct way is to use `<button>`. it's not something we're going to patch directly in bootstrap (i.e. we won't add extra scripting to automagically make those `<a>`s focusable) ... this is something that the authors will have to do themselves if they choose to go through with using that pattern.
